src/main/scala/devices/spi/SPIPhysical.scala

```diff
@@ -59,11 +98,39 @@ class SPIPhysical(c: SPIParamsBase) extends Module {
     Mux(fmt.endian === SPIEndian.MSB, data, Cat(data.toBools))
 
   val rxd = Cat(io.port.dq.reverse.map(_.i))
-  val samples = Seq(rxd(1), rxd(1, 0), rxd)
+  val rxd_delayed = Vec(Seq.fill(io.port.dq.size)(false.B))
+  //0.U(io.port.dq.size.W).toBools
+  //Adding fine-granularity delay buffers on the received data
+
+  val fine_grain_delay_0 = Module(new BlackBoxDelayBuffer())
+  val fine_grain_delay_1 = Module(new BlackBoxDelayBuffer())
+  val fine_grain_delay_2 = Module(new BlackBoxDelayBuffer())
+  val fine_grain_delay_3 = Module(new BlackBoxDelayBuffer())
+  
+  fine_grain_delay_0.io.in := rxd(0)
```

Comment:
then we could loop over the above here rather than writing this out for each one seperately